# ==============================================================================
# FIXING COMMIT DIFF
# ==============================================================================
# File: dom/media/systemservices/objc_video_capture/video_capture_avfoundation.mm
# Commit: f970e7758b64
# Full Hash: f970e7758b6474aaee9ac43d0fcc74f053bed0ab
# Author: Andreas Pehrson <apehrson@mozilla.com>
# Date: 2023-04-21 21:12:46
# Description:
#   Bug 1828722 - Revert 30340541c9a1 - Crash in case VideoCaptureAVFoundation is destroyed unexpectedly. r=webrtc-reviewers,mjf
#   
#   Differential Revision: https://phabricator.services.mozilla.com/D175853
# ==============================================================================

diff -r 1455946414f1 -r f970e7758b64 dom/media/systemservices/objc_video_capture/video_capture_avfoundation.mm
--- a/dom/media/systemservices/objc_video_capture/video_capture_avfoundation.mm	Fri Apr 21 08:42:26 2023 +0000
+++ b/dom/media/systemservices/objc_video_capture/video_capture_avfoundation.mm	Fri Apr 21 08:47:46 2023 +0000
@@ -100,7 +100,6 @@
 
 VideoCaptureAvFoundation::~VideoCaptureAvFoundation() {
   // Must block until capture has fully stopped, including async operations.
-  MOZ_RELEASE_ASSERT(mOkToDestroy);
   mWatchManager.Shutdown();
   StopCapture();
 }